electron/src/ipc-communication.ts

```diff
@@ -29,26 +34,88 @@ class IpcCommunication implements IpcRegister {
                     console.error(err);
                 }
-                files = files
+                const imgsData = fileNames
                     .map((name) => path.join(dir, name))
-                    .filter((imgDir) => {
-                        const imgChunk = readChunk.sync(imgDir, 0, 12);
-                        return imageType(imgChunk)?.mime.startsWith('image') ?? false;
-                    });
-                window.webContents.send('images-found', files);
+                    .filter((name) => this.filterFilesForImages(name))
+                    .map((name) => this.mapToImageData(name));
+                window.webContents.send('images-found', imgsData);
             });
         });
     }
+
+    private getImagesNumber(window: BrowserWindow | null): void {
+        if (!window) {
+            throw new Error('Window was null!');
+        }
+
+        ipcMain.on('get-images-number', (ev, dir: string) => {
+            if (!dir) {
+                window.webContents.send('images-number', 0);
+                return;
+            }
+            fs.readdir(dir, (err, fileNames) => {
+                if (!!err) {
+                    console.error(err);
+                }
+                const num = fileNames
+                    .map((name) => path.join(dir, name))
+                    .filter((name) => this.filterFilesForImages(name)).length;
+                window.webContents.send('images-number', num);
+            });
+        });
+    }
+
+    private getImagesPage(window: BrowserWindow | null): void {
+        if (!window) {
+            throw new Error('Window was null!');
+        }
+
+        ipcMain.on('get-images-page', (_ev, dir: string, currentPage: number, pageSize: number) => {
+            if (!dir) {
+                window.webContents.send('images-page-found', []);
+                return;
+            }
+            fs.readdir(dir, (err, fileNames) => {
+                if (!!err) {
+                    console.error(err);
+                }
+                const start = currentPage * pageSize;
+                const imgsData = fileNames
+                    .map((name) => path.join(dir, name))
+                    .filter((name) => this.filterFilesForImages(name))
+                    .slice(start, start + pageSize)
+                    .map((name) => this.mapToImageData(name));
+                window.webContents.send('images-page-found', imgsData);
+            });
+        });
+    }
+
+    private filterFilesForImages(name: string): boolean {
+        if (fs.lstatSync(name).isDirectory()) {
+            return false;
+        }
+        const imgChunk = readChunk.sync(name, 0, 12);
+        const mime = imageType(imgChunk)?.mime;
+        return !!mime ? mime.includes('image/jpeg') : false;
+    }
+
+    private mapToImageData(name: string): ImageData {
+        const binary = fs.readFileSync(name).toString('binary');
+        const exif = piexif.load(binary);
+        return <ImageData>{
```

Comment:
use recommended _as_ operator for casting

Reply:
Hi Suchar230,

thanks for advice but I will stay with <> operator for casting since there is no functional difference between them and <> has better IDE integration.
Also I haven't found any TS documentation article saying that _as_ operator is recommended (unless you are using some creature like JSX :)). Please email me if you've found some official sources describing this topic - that would be an interesting lecture.

All the best,
Krzysztof

Reply:
The reason for switching to the newer casting is only to resolve ambiguity when using jsx syntax. It does not bring any functional difference in our case, still I believe, however, that going for more widely supported solutions can be a good rule of thumb to use when trying to come up with code convention for this and any other group project 

Let @latusikl decide!

Reply:
Personally I like more _as_ operator but as we do not use JSX syntax I would leave this to individual preferences of developer.